### PR TITLE
Stripped whitespace from the elements

### PR DESCRIPTION
**Pull request by dihora-mayur**
- Updated logic to strip whitespace in languages array/element

### CODE REVIEW

File: talentmap_api/fsbid/services/client.py [hunk 1]
```diff
@@ -546,19 +546,20 @@ def fsbid_languages_to_tmap(languages):
...
             continue
+        if not str(x.get('empl_language')).strip():
+            continue
         r = str(x.get('empl_high_reading', '')).strip()
         s = str(x.get('empl_high_speaking', '')).strip()
         tmap_languages.append({
-            "code": x.get('empl_language_code', None),
-            "language": x.get('empl_language', None),
+            "code": str(x.get('empl_language_code')).strip(),
+            "language": str(x.get('empl_language')).strip(),
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/client.py:555]: We want to do some data checking here and make sure we don't cast to string a `None`

File: talentmap_api/fsbid/services/client.py [hunk 2]
```diff
@@ -546,19 +546,20 @@ def fsbid_languages_to_tmap(languages):
     for x in languages:
         if not x.get('empl_language', None):
             continue
+        if not str(x.get('empl_language')).strip():
```
Comment (elizabeth-jimenez) [inline talentmap_api/fsbid/services/client.py:549]: we may want to move this strip to line 559

File: talentmap_api/fsbid/services/client.py (hunk 1 above)
Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/client.py:555]: something like this `a = str(x.get('a')).strip() if x.get('a') else x.get('a') or None` should cover all our bases

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/client.py:555]: image

Comment (elizabeth-jimenez) [reply talentmap_api/fsbid/services/client.py:555]: i'd double check the `or None` tho - we weren't already handling that, although we do want to, so i'd just make sure the FE is alright with it.

**Comment by elizabeth-jimenez** [comment]
Great job with your first attempt at python :)

**Comment by SidelineCory24** [comment]
[FE PR](https://github.com/MetaPhase-Consulting/State-TalentMAP/pull/2319)

**Comment by SidelineCory24** [comment]
@dihora-mayur looks like Elizabeth left two more changes that need to be made, then this pr should be gtg.

**Comment by dihora-mayur** [comment]
Not sure how I missed this. Just committed these changes.

From: SidelineCory24 ***@***.***>
Date: Thursday, September 22, 2022 at 2:53 PM
To: MetaPhase-Consulting/State-TalentMAP-API ***@***.***>
Cc: Dihora, Mayur [USA] ***@***.***>, Mention ***@***.***>
Subject: [External] Re: [MetaPhase-Consulting/State-TalentMAP-API] Stripped whitespace from the elements (PR #717)

@dihora-mayur<https://urldefense.com/v3/__https:/github.com/dihora-mayur__;!!May37g!Lxk1e46Ub8XLM9oN1feEeP8SzhUCniltqy4iqe3KSImYgQn93A35vQHSjwzrvoN5leIwjEWyfhucYR-yGM-0fv9wdg$> looks like Elizabeth left two more changes that need to be made, then this pr should be gtg.

File: talentmap_api/fsbid/services/client.py (hunk 2 above)
Comment (dihora-mayur) [reply talentmap_api/fsbid/services/client.py:549]: I added strip to line 559 but I kept this one as it still is relevant here to stop going forward and return None if the empl_lang element initially has a whitespace only. There may be a way to condense lines 547 and 549 though?

Comment (SidelineCory24) [reply talentmap_api/fsbid/services/client.py:549]: you can combine this way using `or`
```
if not x.get('empl_language', None) or not str(x.get('empl_language')).strip():
            continue
```

**Comment by dihora-mayur** [comment]
😃 I was hung up on trying to condense it within the get(). But this works too. I updated the change directly through git.

From: SidelineCory24 ***@***.***>
Date: Friday, September 23, 2022 at 9:20 AM
To: MetaPhase-Consulting/State-TalentMAP-API ***@***.***>
Cc: Dihora, Mayur [USA] ***@***.***>, Mention ***@***.***>
Subject: [External] Re: [MetaPhase-Consulting/State-TalentMAP-API] Stripped whitespace from the elements (PR #717)

@SidelineCory24 commented on this pull request.

________________________________

In talentmap_api/fsbid/services/client.py<https://urldefense.com/v3/__https:/github.com/MetaPhase-Consulting/State-TalentMAP-API/pull/717*discussion_r978655228__;Iw!!May37g!NZg1N6q0StLmQGecyYwFE77kYyRdP-URucF43Pujmkp2X98e1x4LUdvUuORUUEldkDlQ3rICqd-XxFYfPdw8slsCog$>:

> @@ -546,19 +546,20 @@ def fsbid_languages_to_tmap(languages):

     for x in languages:

         if not x.get('empl_language', None):

             continue

+        if not str(x.get('empl_language')).strip():

you can combine this way using or

if not x.get('empl_language', None) or not str(x.get('empl_language')).strip():

            continue

**Comment by SidelineCory24** [comment]
> 

we're manipulating the get differently with `str()` in one check so we'll want to keep them separate, nice work!

**Review by SidelineCory24 — APPROVED** [review]
Looks good to go! 🚀

**Comment by SidelineCory24** [comment]
@elizabeth-jimenez I remember you mentioning the `pylint` is failing. Would you be able to override this to merge the pr or is something else needed. I can't do the force merge on my end. Thanks!

**Comment by elizabeth-jimenez** [comment]
> @elizabeth-jimenez I remember you mentioning the `pylint` is failing. Would you be able to override this to merge the pr or is something else needed. I can't do the force merge on my end. Thanks!

Did y'all dbl check the FE for the new None default?

**Comment by SidelineCory24** [comment]
> > @elizabeth-jimenez I remember you mentioning the `pylint` is failing. Would you be able to override this to merge the pr or is something else needed. I can't do the force merge on my end. Thanks!
> 
> Did y'all dbl check the FE for the new None default?

I can double check again, it says outdated for that comment so I though it was already addressed.

**Comment by elizabeth-jimenez** [comment]
> > > @elizabeth-jimenez I remember you mentioning the `pylint` is failing. Would you be able to override this to merge the pr or is something else needed. I can't do the force merge on my end. Thanks!
> > 
> > 
> > Did y'all dbl check the FE for the new None default?
> 
> I can double check again, it says outdated for that comment so I though it was already addressed.

Right, that means the change was made 😅 but not that the change was tested

**Comment by dihora-mayur** [comment]
> > > > @elizabeth-jimenez I remember you mentioning the `pylint` is failing. Would you be able to override this to merge the pr or is something else needed. I can't do the force merge on my end. Thanks!
> > > 
> > > 
> > > Did y'all dbl check the FE for the new None default?
> > 
> > 
> > I can double check again, it says outdated for that comment so I though it was already addressed.
> 
> Right, that means the change was made 😅 but not that the change was tested

So i had tested it previously and just re-tested all of these right now and the column displays None in all these instances:
if languages array itself is sent as [ ] or [ { } ]. If there is key value pairs but empl_language_code = ‘’ or ' ’ or this key doesn’t exist at all, empl_language = ‘’ or ' ' or this key doesn’t exist at all

**Pull request merged**